### ISSUE TITLE
Postmark v2.5.0 & v2.5.1 Has Cryptic XMLHttpRequest Issues

### ISSUE THREAD

**Issue by zjr** [wildbit/postmark.js#61]
When using `postmark.TemplatedMessage` and `sendEmailBatchWithTemplates` I get an odd error without a lot of description. Downgrading to 2.3.5 fixes this issue. The problem is present in both 2.5.0 & 2.5.1.

Message:

```
PostmarkError: The string did not match the expected pattern.
    at ErrorHandler.Object.<anonymous>.ErrorHandler.buildGeneralError (/Users/zjr/Developer/Current/csns/node_modules/postmark/src/client/ErrorHandler.ts:37:16)
    at ErrorHandler.Object.<anonymous>.ErrorHandler.buildRequestError (/Users/zjr/Developer/Current/csns/node_modules/postmark/src/client/ErrorHandler.ts:23:25)
    at /Users/zjr/Developer/Current/csns/node_modules/postmark/src/client/BaseClient.ts:116:41
```

**Comment by ibalosh** [comment]
Hi @zjr 

we could not reproduce the issue yet. Could you share codesnipet without credentials/auth details that you used and got the error?

Also if you could share typescript, nodejs version you use that would be great.

Thanks!

**Comment by zjr** [comment]
Hi Igor, sorry I didn't think to include the Node version, I'm on v12.16.0 (no TS). I'm very busy this week but I'll try to come up with a test case soon.

**Comment by ibalosh** [comment]
Hey,

no worries, tx!

**Comment by ibalosh** [comment]
Hi,

we could not reproduce this issue, and haven't seen any other reports for it. Since then the error handling part had updates too.

I am closing the issue. In case you can still see it with latest (2.5.3) library version, feel free to send over details and we will check it out.

Igor

**Comment by hequ** [comment]
Just commenting here if anyone encounters this in the future. I got this same error when running tests with jest and using postmark client 2.7.4. Jest runs by default in `jsdom` mode so it has a window. When I changed jest to run in `node`, this error went away. Apparently, Axios is doing something funny in a browser context.

If you are interested, there are a couple of related errors which might help to understand the core reason:

https://github.com/facebook/jest/issues/5737
https://github.com/axios/axios/issues/1180
